src/python/pants/bin/goal_runner.py

```diff
@@ -46,21 +60,12 @@ def __init__(self, root_dir, options, build_config, run_tracker, reporting, grap
     self._explain = self._global_options.explain
     self._kill_nailguns = self._global_options.kill_nailguns
 
-  def _determine_v1_goals(self, address_mapper, options):
+  def _determine_v1_goals(self, options: Options) -> List[Goal]:
     """Check and populate the requested goals for a given run."""
     v1_goals, ambiguous_goals, _ = options.goals_by_version
-    requested_goals = v1_goals + ambiguous_goals
-
-    spec_parser = CmdLineSpecParser(self._root_dir)
-
-    for goal in requested_goals:
-      if address_mapper.is_valid_single_address(spec_parser.parse_address_spec(goal)):
-        logger.warning("Command-line argument '{0}' is ambiguous and was assumed to be "
-                       "a goal. If this is incorrect, disambiguate it with ./{0}.".format(goal))
```

Comment:
I couldn't figure out how to get this to ever trigger. The `arg_splitter` already handles all ambiguity between ambiguous goals and has several heuristics to disambiguate, for example, between the directory `fmt` and the goal `fmt`. 

Due to those checks, I couldn't get this code to ever trigger. So, I removed it for less complexity.